src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/generalizing-ownership.md

```diff
@@ -0,0 +1,68 @@
+---
+minutes: 5
+---
+
+# Generalizing Ownership: Lifetimes Refresher
+
+The logic of the borrow checker, while modelled off "memory ownership", can be
+abstracted away from that use case to model other problems where we want to
+prevent API misuse.
+
+```rust,editable
+// An internal data type to have something to hold onto.
+pub struct Internal;
+// The "outer" data.
+pub struct Data(Internal);
+
+fn shared_use(value: &Data) -> &Internal {
+    &value.0
+}
+fn exclusive_use(value: &mut Data) -> &mut Internal {
+    &mut value.0
+}
+fn deny_future_use(value: Data) {}
+
+fn main() {
+    let mut value = Data(Internal);
+    let shared = shared_use(&value);
+    // let exclusive = exclusive_use(&mut value); // ❌🔨
+    let shared_again = &shared;
+    deny_future_use(value);
+    // let shared_again_again = shared_use(&value); // ❌🔨
```

Comment:
I find this code example more clear than before, thank you!

I do feel though that 3 ideas (what each parameter passing mode achieves) are too cramped in one function and interfere with each other. Could we split this into 3 pseudo-main functions? It does not matter that the code would not have an actual main, there is nothing runnable here anyway.